src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/QualByDepth.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * <h3>Caveat</h3>
+ * <p>This annotation can only be calculated for sites for which at least one sample was genotyped as carrying a variant allele.</p>
+ *
+ * <h3>Related annotations</h3>
+ * <ul>
+ *     <li><b><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_Coverage.php">Coverage</a></b> gives the filtered depth of coverage for each sample and the unfiltered depth across all samples.</li>
+ *     <li><b><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.php">DepthPerAlleleBySample</a></b> calculates depth of coverage for each allele per sample (AD).</li>
+ * </ul>
+ */
+public final class QualByDepth extends InfoFieldAnnotation {
+
+    @VisibleForTesting
+    static final double MAX_QD_BEFORE_FIXING = 35;
+
+    @VisibleForTesting
+    static final double IDEAL_HIGH_QD = 30;
+    private static final double JITTER_SIGMA = 3;
+
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                        final VariantContext vc,
+                                        final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap) {
+        Utils.nonNull(vc);
+        if ( !vc.hasLog10PError() ) {
+            return null;
+        }
+
+        final GenotypesContext genotypes = vc.getGenotypes();
+        if ( genotypes == null || genotypes.isEmpty() ) {
+            return null;
+        }
+
+        int depth = 0;
+        int ADrestrictedDepth = 0;
+
+        for ( final Genotype genotype : genotypes ) {
+            // we care only about variant calls with likelihoods
+            if ( !genotype.isHet() && !genotype.isHomVar() ) {
+                continue;
+            }
+
+            // if we have the AD values for this sample, let's make sure that the variant depth is greater than 1!
+            // TODO -- If we like how this is working and want to apply it to a situation other than the single sample HC pipeline,
+            // TODO --  then we will need to modify the annotateContext() - and related - routines in the VariantAnnotatorEngine
+            // TODO --  so that genotype-level annotations are run first (to generate AD on the samples) and then the site-level
+            // TODO --  annotations must come afterwards (so that QD can use the AD).
```

Comment:
Yeah, Ron definitely made changes here recently.  You might want to pull from gsa-unstable or whatever is equivalent.